pkg/scheduler/framework/runtime/framework.go

```diff
@@ -655,14 +655,16 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 	var result *framework.PreFilterResult
 	var pluginsWithNodes []string
 	logger := klog.FromContext(ctx)
-	logger = klog.LoggerWithName(logger, "PreFilter")
-	// TODO(knelasevero): Remove duplicated keys from log entry calls
-	// When contextualized logging hits GA
-	// https://github.com/kubernetes/kubernetes/issues/111672
-	logger = klog.LoggerWithValues(logger, "pod", klog.KObj(pod))
+	verboseLogs := logger.V(4).Enabled()
+	if verboseLogs {
+		logger = klog.LoggerWithName(logger, "PreFilter")
```

Comment:
Overall, the change LGTM, but I wonder if we could do better by delegating to klog.

For example, we could say something like:

```golang
logger = klog.LoggerWithNameAt(logger, "PreFilter", 4)
```

which would just return the same logger if the level is less than 4.

However, I'm not sure what to do about `klog.NewContext`. Should we have a function `klog.ContextAndLoggerWithNameAt`? It sounds ugly, and we would also need a counterpart for `WithLabels` (so 4 new functions in total). Maybe `klog.LoggerWithNameAt` can return a boolean that says whether it created a new logger?

In any case, this discussion is beyond what we can cherry-pick, so we should merge like this anyways. Just food for thought.

Reply:
uhm... maybe it could be a `loggingutils` package within `pkg/scheduler`, where even the constant 4 could be hidden from the developer.